### PR TITLE
feat: add steps component

### LINKED ISSUE #424: Steps Component
**Is your feature request related to a problem? Please describe.**

As a developer I'll need a stepper component.

**Describe the solution you'd like**

A steps component styled like the following:
http://ux-document-lnx/~designer/sbg-ux/2.1.0/components/organisms/wizard.html

Themed. Accessible.

### PR DESCRIPTION
**Pull request by BogdanV3**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows semantic commit message guidelines
- [x] The changes are documented in component docs and changelog
- [x] The ESLint plugin has been updated if a new component is added
- [x] Test have been added or modified, if appropriate
- [x] Has been verified locally

**What kind of change does this PR introduce?** 

Closes #424 

**Does this PR introduce a breaking change?** 

**Other information**:

### CODE REVIEW

File: framework/styles/theme/default.scss [hunk 1]
```diff
@@ -160,6 +160,14 @@ $atomic-default: (
     "selection-bg-disabled": map-get($grey, "lighten-2"),
     "handle-border-disabled": map-get($grey, "lighten-2")
   ),
+  "steps": (
+    "step-icon-bg-visited": transparent,
+    "step-icon-color-visited": map-get($grey, "lighten-5"),
+    "step-icon-bg": transparent,
+    "step-icon-color": map-get($grey, "lighten-5"),
+    "step-icon-border-color": rgba(map-get($grey, "lighten-2"), $alpha-3),
+    "step-line-color": rgba(map-get($grey, "lighten-2"), $alpha-3)
```
Comment (frattaro) [inline framework/styles/theme/default.scss:169]: These colors need to match the original values if the sass variables:
```
$step-icon-bg-visited: $base-inverse-color; // copy value from "base" section above
$step-icon-color-visited: map-get($grey, "darken-4");
$step-icon-bg: $base-inverse-color; // copy value from "base" section above
$step-icon-color: map-get($grey, "darken-4");
$step-icon-border-color: map-get($grey, "lighten-2");
$step-line-color: map-get($grey, "lighten-2");
```

might not need `step-icon-bg-visited` and `step-icon-bg` if those theme values can be replaced with `"base", "inverse-color"`

**Review by frattaro — CHANGES_REQUESTED** [review]
The last `a-step` needs it's divider `display: none`'d to match atomic.

I think the last thing is to add an entry in `/framework/changelog.mdx` like
```
## 3.2.0 (pending)

- Added `AStepper` component
```

File: framework/components/AStepper/AStepper.spec.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    );
+    cy.get(
+      "#vertical-stepper-with-icon + .playground .a-steps--orientation-vertical"
+    )
+      .get(".a-step")
+      .eq(0)
+      .get(".a-step__icon__checkmark")
+      .should("be.visible");
+  });
+});
```
Comment (frattaro) [inline framework/components/AStepper/AStepper.spec.js:65]: Can you add one more test that takes screenshots? And then after that's committed, manually run the `Update Snapshots` github workflow against this branch -- described here: https://github.com/cisco-sbg-ui/atomic-react/issues/987#issuecomment-955607568

Comment (frattaro) [reply framework/components/AStepper/AStepper.spec.js:77]: This also needs to flip the dusk-mode switch on the page and re-take the screen shots

File: framework/components/AStepper/AStepper.mdx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+## Stepper Props
+
+All AStepper components (AStepper, AStep, AStepTitle, AStepDescription) inherit passed props.
+
+<Props of="AStepper" />
+
+## Step Props
+
+<Props of="AStep" />
```
Comment (frattaro) [inline framework/components/AStepper/AStepper.mdx:202]: Can you add the props for AStepTitle and AStepDescription here too?

**Comment by frattaro** [comment]
Looks like the backgrounds need removed from the titles:
image

**Comment by frattaro** [comment]
And disabled steps should have the number be the same color as the circle around it

File: framework/components/AStepper/AStepper.scss [hunk 4]
```diff
@@ -1,5 +1,51 @@
...
+      background-color: map-deep-get($theme, "steps", "step-line-color");
+    }
+    &__visited {
+      .a-step__divider {
+        background-color: $color-good;
+        border: 0px;
+        height: 1px;
+      }
+    }
+    &:before {
```
Comment (frattaro) [inline framework/components/AStepper/AStepper.scss:28]: Now that we have an element for the line, I don't believe the `:before` line styles need to exist

File: framework/components/AStepper/AStepper.scss [hunk 5]
```diff
@@ -1,5 +1,51 @@
...
+    .a-step {
+      border: map-get($theme, "base", "border-width") solid
+        map-get($theme, "control", "stroke-color");
+    }
+  }
+}
+
+$steps-min-width: 0;
 $steps-max-width: 1000px;
 $steps-max-width-vertical: 300px;
```
Comment (frattaro) [inline framework/components/AStepper/AStepper.scss:50]: The steps' width should be determined by layout containers, not inherent to the component -- can you please remove these and their associated rule properties?

File: framework/components/AStepper/AStepper.scss [hunk 6]
```diff
@@ -34,9 +75,7 @@ $step-line-top: 22px;
 $step-line-vertical-top: 44px;
 $step-line-left: 44px;
 $step-line-vertical-left: 22px;
```
Comment (frattaro) [inline framework/components/AStepper/AStepper.scss:77]: Can you please remove all of these scss variables? They're potential collisions, don't have proper defaults, and even if they had proper defaults we don't yet have a proper solution for documenting scss variables

**Review by frattaro — COMMENTED** [review]
`&--type-extended` doesn't look like it's used anywhere so we should remove the unused code

File: framework/components/AStepper/AStepper.scss (hunk 6 above)
Comment (BogdanV3) [reply framework/components/AStepper/AStepper.scss:77]: only this 3? or all of them?

Comment (frattaro) [reply framework/components/AStepper/AStepper.scss]: all of them

**Review by frattaro — APPROVED** [review]
This is beautiful. Please squash-merge at your pleasure -- feel free to clear out the commit body

**Pull request merged**